projects/storefrontlib/src/cms-structure/page/component/component-wrapper.directive.spec.ts

```diff
@@ -177,7 +186,39 @@ describe('ComponentWrapperDirective', () => {
         );
       });
 
-      it('should add smartedit contract if app launch in smart edit', () => {
+      describe('events', () => {
+        it('should dispatch ComponentCreateEvent on creation', () => {
+          spyOn(eventService, 'dispatch').and.callThrough();
+          fixture.detectChanges();
+
+          const el = fixture.debugElement;
+          const compEl = el.query(By.css('cx-test')).nativeElement;
+
+          expect(eventService.dispatch).toHaveBeenCalledWith(
+            {
+              typeCode: 'cms_typeCode',
+              id: 'test_uid',
+              host: compEl,
+            } as ComponentEvent,
+            ComponentCreateEvent
+          );
+        });
+
+        it('should dispatch ComponentDestroyEvent on creation', () => {
+          spyOn(eventService, 'dispatch').and.callThrough();
+          fixture.detectChanges();
+          fixture.destroy();
+          expect(eventService.dispatch).toHaveBeenCalledWith(
+            {
+              typeCode: 'cms_typeCode',
+              id: 'test_uid',
+            } as ComponentEvent,
+            ComponentDestroyEvent
+          );
+        });
```

Comment:
Maybe we could also subscribe to event?

Reply:
That's not the concern of the comp wrapper imo